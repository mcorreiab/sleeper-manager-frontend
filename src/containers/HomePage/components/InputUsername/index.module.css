.inputUsername {
  display: block;
  width: 100%;
  padding: 0.75rem 1rem;
  border-radius: 0.5rem;
  font-size: 1rem;
  border: 1px solid rgb(46, 46, 46);
  outline: none;
}

.inputUsername:focus {
  border: 2px solid #0073cd;
}

.inputUsernameError {
  border-color: #e30202;
}

.errorMessage {
  color: #e30202;
  padding-left: 1rem;
  margin-bottom: 1rem;
  margin-top: 0.25rem;
}

.button {
  background-color: #fa7820;
  border-radius: 0.5rem;
  width: 100%;
  height: auto;
  padding: 0.75rem 1rem;
  font-size: 1rem;
  border: 0;
  color: white;
  font-weight: bold;
  outline: none;
}
