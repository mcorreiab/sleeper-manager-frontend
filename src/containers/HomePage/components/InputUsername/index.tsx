import React, { FormEvent, ChangeEvent, FunctionComponent } from "react";
import styles from "./index.module.css";

interface Props {
  className: string;
  onFormSubmit: (event: FormEvent<HTMLFormElement>) => void;
  username: string;
  onChangeUsername: (event: ChangeEvent<HTMLInputElement>) => void;
  isUsernameMissing: boolean;
}

const inputUsername: FunctionComponent<Props> = ({
  className,
  onFormSubmit,
  username,
  onChangeUsername,
  isUsernameMissing,
}) => {
  const inputUsernameClasses = [styles.inputUsername];
  let usernameMissing: string = null;
  if (isUsernameMissing) {
    inputUsernameClasses.push(styles.inputUsernameError);
    usernameMissing = "Should inform an username";
  }

  return (
    <div className={className}>
      <form onSubmit={onFormSubmit}>
        <input
          className={inputUsernameClasses.join(" ")}
          placeholder="Insert your username here"
          value={username}
          onChange={onChangeUsername}
        />
        <p className={styles.errorMessage}>{usernameMissing}</p>
        <input className={styles.button} type="submit" value="GO" />
      </form>
    </div>
  );
};
export default inputUsername;
