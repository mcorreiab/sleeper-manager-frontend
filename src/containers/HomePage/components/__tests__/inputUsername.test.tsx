import { render } from "@testing-library/react";
import userEvent from "@testing-library/user-event";
import renderer from "react-test-renderer";
import InputUsername from "../inputUsername";

it("should render the username input without username", () => {
  const tree = renderer
    .create(
      <InputUsername
        onFormSubmit={() => {}}
        onChangeUsername={() => {}}
        username=""
      />
    )
    .toJSON();
  expect(tree).toMatchSnapshot();
});

it("should render the username input with an username", () => {
  const tree = renderer
    .create(
      <InputUsername
        onFormSubmit={() => {}}
        onChangeUsername={() => {}}
        username="username"
      />
    )
    .toJSON();
  expect(tree).toMatchSnapshot();
});

it("should render the username input with user missing error state", () => {
  const tree = renderer
    .create(
      <InputUsername
        onFormSubmit={() => {}}
        onChangeUsername={() => {}}
        username="username"
        isUsernameMissing
      />
    )
    .toJSON();
  expect(tree).toMatchSnapshot();
});

it("should render the username input with user invalid error state", () => {
  const tree = renderer
    .create(
      <InputUsername
        onFormSubmit={() => {}}
        onChangeUsername={() => {}}
        username="username"
        isUsernameInvalid
      />
    )
    .toJSON();
  expect(tree).toMatchSnapshot();
});

it("should call a callback when typing an username", async () => {
  const onChangeUsername = jest.fn();
  const { getByPlaceholderText } = render(
    <InputUsername
      onFormSubmit={() => {}}
      onChangeUsername={onChangeUsername}
      username=""
    />
  );

  const input = getByPlaceholderText("Insert your username here");
  await userEvent.type(input, "username");
  expect(onChangeUsername).toHaveBeenCalled();
});

it("should call a callback when click on go button", async () => {
  const onFormSubmit = jest.fn();
  const { getByText } = render(
    <InputUsername
      onFormSubmit={onFormSubmit}
      onChangeUsername={() => {}}
      username=""
    />
  );

  const input = getByText("GO");
  await userEvent.click(input);
  expect(onFormSubmit).toHaveBeenCalled();
});
