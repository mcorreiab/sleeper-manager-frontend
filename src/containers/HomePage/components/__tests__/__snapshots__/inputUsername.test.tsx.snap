// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render the username input with an username 1`] = `
<div>
  <form
    onSubmit={[Function]}
  >
    <input
      className="block w-full py-3 px-4 rounded-lg text-base border 
  border-solid border-[#2E2E2E] text-black outline-none
  focus:border-2 focus:border-[#0073cd] invalid:border-[#e30202]"
      onChange={[Function]}
      placeholder="Insert your username here"
      required={false}
      type="text"
      value="username"
    />
    <p
      className="text-[#e30202] pl-4 mb-4 mt-1"
    />
    <input
      className="bg-sm-orange rounded-lg w-full h-auto py-3 px-4 
        text-base border-0 text-white font-bold outline-none"
      type="submit"
      value="GO"
    />
  </form>
</div>
`;

exports[`should render the username input with user invalid error state 1`] = `
<div>
  <form
    onSubmit={[Function]}
  >
    <input
      className="block w-full py-3 px-4 rounded-lg text-base border 
  border-solid border-[#2E2E2E] text-black outline-none
  focus:border-2 focus:border-[#0073cd] invalid:border-[#e30202]"
      onChange={[Function]}
      placeholder="Insert your username here"
      required={false}
      type="text"
      value="username"
    />
    <p
      className="text-[#e30202] pl-4 mb-4 mt-1"
    >
      User not found
    </p>
    <input
      className="bg-sm-orange rounded-lg w-full h-auto py-3 px-4 
        text-base border-0 text-white font-bold outline-none"
      type="submit"
      value="GO"
    />
  </form>
</div>
`;

exports[`should render the username input with user missing error state 1`] = `
<div>
  <form
    onSubmit={[Function]}
  >
    <input
      className="block w-full py-3 px-4 rounded-lg text-base border 
  border-solid border-[#2E2E2E] text-black outline-none
  focus:border-2 focus:border-[#0073cd] invalid:border-[#e30202]"
      onChange={[Function]}
      placeholder="Insert your username here"
      required={true}
      type="text"
      value="username"
    />
    <p
      className="text-[#e30202] pl-4 mb-4 mt-1"
    >
      Should inform an username
    </p>
    <input
      className="bg-sm-orange rounded-lg w-full h-auto py-3 px-4 
        text-base border-0 text-white font-bold outline-none"
      type="submit"
      value="GO"
    />
  </form>
</div>
`;

exports[`should render the username input without username 1`] = `
<div>
  <form
    onSubmit={[Function]}
  >
    <input
      className="block w-full py-3 px-4 rounded-lg text-base border 
  border-solid border-[#2E2E2E] text-black outline-none
  focus:border-2 focus:border-[#0073cd] invalid:border-[#e30202]"
      onChange={[Function]}
      placeholder="Insert your username here"
      required={false}
      type="text"
      value=""
    />
    <p
      className="text-[#e30202] pl-4 mb-4 mt-1"
    />
    <input
      className="bg-sm-orange rounded-lg w-full h-auto py-3 px-4 
        text-base border-0 text-white font-bold outline-none"
      type="submit"
      value="GO"
    />
  </form>
</div>
`;
